src/dune_engine/fs_cache.ml

```diff
@@ -32,36 +24,20 @@ let read { sample; cache; _ } path =
 
 let evict { cache; _ } path = Path.Outside_build_dir.Table.remove cache path
 
-module Update_result = struct
-  type t =
-    | Skipped (* No need to update a given entry because it has no readers *)
-    | Updated of { changed : bool }
-
-  let combine x y =
-    match (x, y) with
-    | Skipped, res | res, Skipped -> res
-    | Updated { changed = x }, Updated { changed = y } ->
-      Updated { changed = x || y }
-
-  let empty = Skipped
-
-  let to_dyn = function
-    | Skipped -> Dyn.Variant ("Skipped", [])
-    | Updated { changed } ->
-      Dyn.Variant ("Updated", [ Dyn.Record [ ("changed", Dyn.Bool changed) ] ])
-end
-
-let update { sample; cache; equal; update_hook; _ } path =
-  match Path.Outside_build_dir.Table.find cache path with
-  | None -> Update_result.Skipped
-  | Some old_result -> (
-    update_hook path;
```

Comment:
What happened to the `update_hook`? It looks like we're no longer running it. Is that right?

Reply:
Yup. B/c it was only used to sync the two caches. That's no longer necessary as there's only one cache.

Reply:
We run `refresh_without_removing_write_permissions` now but we used to run `invalidate_cached_timestamp`. It seems like possibly this is an improvement but I'm not sure. Is it?

Reply:
Hmm, good point. Let me double check.

Reply:
I had a look and the code is equivalent. Previously, `invalidate_cached_timestamp` would update the entry in the table and the subsequent `sample` would refresh it. In this PR, I just skip the invalidate because it's going to to be overwritten by the refresh

Reply:
OK, thanks!